### PR TITLE
Added EntityItemDamageEvent

### PR DESCRIPTION
**Pull request by Machine-Maker**
Barty made a comment that he was trying to cancel the durability damage armor took from thorns on armor stands. This event should cover that, and, well all other cases of items on other entities that took durability damage.

### CODE REVIEW

**Review by MiniDigger — APPROVED** [review]
lgtm

**Comment by Machine-Maker** [comment]
ugh

File: Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
++
++/**
++ * Called when an item on or used by an entity takes durability damage as a result of being hit/used.
++ *
++ * <p>
++ *     NOTE: default vanilla behaviour dictates that armor/tools picked up by
++ *     mobs do not take damage (except via Thorns).
++ * </p>
++ */
++public class EntityItemDamageEvent extends EntityEvent implements Cancellable {
```
Comment (kashike) [inline Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]:
```suggestion
+public class EntityDamageItemEvent extends EntityEvent implements Cancellable {
```
?

**Review by kashike — DISMISSED** [review]

File: Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch (hunk 1 above)
Comment (Machine-Maker) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: Doesn't that sound like its fired when an entity damages an item? Which isn't really what this is.

Comment (BillyGalbreath) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: EntityItemDamageEvent makes it sound like I can cancel the damage to Item entities (like throwing an item into fire)

Comment (Machine-Maker) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: There already is a PlayerItemDamageEvent which is called when this stuff happens to a player.

Comment (kashike) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: @BillyGalbreath's comment is why I suggested this change.

Comment (Machine-Maker) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: But then the naming isn't consistent with the already existing player event?

Comment (kashike) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: Names not being consistent does not worry me.

**Comment by Machine-Maker** [comment]
`Rebased after the NMS repackage`

**Comment by Machine-Maker** [comment]
thanks, but no thanks

File: Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch (hunk 1 above)
Comment (Machine-Maker) [reply Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch]: Ok, changed the name

**Review by Proximyst — APPROVED** [review]

**Comment by 22s** [comment]
Could the slot of the item damaged be added to this event?

**Pull request merged**